shared/src/main/kotlin/org/javacs/kt/classpath/DefaultClassPathResolver.kt

```diff
@@ -1,15 +1,19 @@
 package org.javacs.kt.classpath
 
 import org.javacs.kt.LOG
+import org.javacs.kt.storage.Storage
 import java.nio.file.Path
 import java.nio.file.PathMatcher
 import java.nio.file.FileSystems
 
-fun defaultClassPathResolver(workspaceRoots: Collection<Path>): ClassPathResolver =
-    WithStdlibResolver(
-        ShellClassPathResolver.global(workspaceRoots.firstOrNull())
-            .or(workspaceRoots.asSequence().flatMap(::workspaceResolvers).joined)
-    ).or(BackupClassPathResolver)
+fun defaultClassPathResolver(workspaceRoots: Collection<Path>, storage: Storage?): ClassPathResolver =
```

Comment:
Perhaps as an optional parameter to make the call site a bit more concise:

```suggestion
fun defaultClassPathResolver(workspaceRoots: Collection<Path>, storage: Storage? = null): ClassPathResolver =
```